app/forms/placements_form.rb

```diff
@@ -1,7 +1,118 @@
 # frozen_string_literal: true
 
-class PlacementsForm < TraineeForm
-  def compute_fields
-    {}
+class PlacementsForm
+  include ActiveModel::Model
+
+  attr_accessor :trainee, :placement_ids
+
+  # validate :must_have_minimum_number_of_placements
```

Comment:
I've left this out for now because otherwise it breaks the confirmation page - you can't submit the form with the checkbox unselected and less than two placements. In other words you can't say I'll come back and do this later.

I'm assuming the validation logic should be here rather than some other separate confirmation validator? If so the issue I have is that I want to invoke validations only when the user selects ticks the _Completed_ checkbox but the form object doesn't know anything about that checkbox and there's no simple way to pass it into the `PlacementsForm`.

Reply:
Ummm you have the `trainee`

So you can now have figure out if you need to validate it altogether

`trainee.placement_detail`  or 
`trainee.progress.placement`